### PR TITLE
Build & Deploy docs GitHub action: auto delete cancelled

### PR DESCRIPTION
**Pull request by Theodlz**
Based on a suggestion from @stefanv, this PR is an attempt at using the github action from https://github.com/MercuryTechnologies/delete-cancelled-runs.

The problem we are facing is that the github action we rely on to build&deploy the docs already cancels previous workflows that are still running, except that when workflow are just cancelled they make the overall test suite appear as failed. Deleting cancelled workflows makes a lot of sense here, and prevents the tests from appearing as failed.

Also, I made small changes to the `labeler.yaml` config, because pretty much all of our PRs ended up with this `needs-migrations?` label. I think that it needs merging for that behavior to stop.

### CODE REVIEW

**Comment by Theodlz** [comment]
Btw this makes me wonder if we want to do the same for all the unit tests (auto cancel + maybe delete) within a single PR (and not on main).

Cancelling a PRs actions that are still running once we merge it (which happens with the dependabot stuff sometimes) would be great too.

File: .github/workflows/build-and-deploy-docs.yaml
Comment (stefanv) [inline .github/workflows/build-and-deploy-docs.yaml:17]: Better to pin to the hash here. See https://scientific-python.org/specs/spec-0008/#pin-github-actions-release-workflows-to-their-full-release-commit-shas

**Review by stefanv — APPROVED** [review]
Approved, pending pinning to hashed action.

File: .github/workflows/build-and-deploy-docs.yaml
Comment (Theodlz) [reply .github/workflows/build-and-deploy-docs.yaml:17]: pinning any version failed, so yeah I'll try the hash anyway hahaha

Comment (Theodlz) [reply .github/workflows/build-and-deploy-docs.yaml]: @stefanv I suppose that we don't actually need a separate job like in their docs, but that the step could be the first step of our existing build&deploy docs job?

Comment (stefanv) [reply .github/workflows/build-and-deploy-docs.yaml]: Yes, I think that's fine.

**Pull request merged**
